CORPSE_deriv.py

```diff
@@ -170,18 +180,27 @@ def CORPSE_deriv(SOM,T,theta,Ndemand,Ctransfer,params,claymod=1.0):
             dmicrobeC[mt] = (carbon_supply[mt] - microbeTurnover[mt]) * loc_Clim
             dmicrobeN[mt] = dmicrobeC[mt] / params['CN_microbe'][mt]
 
-    # If mycorrhizal fungi transfer too much N to plants (exceeding plant N demands), then mycorrhizal N acquisition
+    # If mycorrhizal fungi transfer too much N to plants (N_int pool exceeding 2*Ndemand), then mycorrhizal N acquisition
     # is decreased accordingly. This will not be needed once coupled to a plant growth model.
     Ntransfer = max(0.0,CN_imbalance_term['ECM']) + max(0.0,CN_imbalance_term['AM'])
-    if Ntransfer>Ndemand:
-       Nmining_d = (Ntransfer-Ndemand)*max(0.0,CN_imbalance_term['ECM'])/Ntransfer
-       for t in chem_types:
-           if nitrogen_supply['ECM']>0.0:
-              Nmining[t+'N'] = Nmining[t+'N']*(nitrogen_supply['ECM']-Nmining_d)/nitrogen_supply['ECM']
-       Nscavenging_d = Ntransfer-Ndemand-Nmining_d
-       nitrogen_supply['AM'] += -Nscavenging_d
-    else:
-       print('!! Unbalanced N budget for the plants')
+    # if Ntransfer>Ndemand:
+    #    Nmining_d = (Ntransfer-Ndemand)*max(0.0,CN_imbalance_term['ECM'])/Ntransfer
+    #    for t in chem_types:
+    #        if nitrogen_supply['ECM']>0.0:
+    #           Nmining[t+'N'] = Nmining[t+'N']*(nitrogen_supply['ECM']-Nmining_d)/nitrogen_supply['ECM']
+    #    Nscavenging_d = Ntransfer-Ndemand-Nmining_d
+    #    nitrogen_supply['AM'] += -Nscavenging_d
+    if SOM['Int_N']+Ntransfer-Nlitter>2*Ndemand:
```

Comment:
This is mixing up units, since the units of all pools in `SOM` should be stock (e.g. g N) while `Ntransfer` and `Ndemand` are rates (g N/year). This could be resolved by applying a rate constant parameter to the intermediate pools as I suggested above.